src/main/scala/it/unibo/authsim/library/dsl/UserProvider.scala

```diff
@@ -0,0 +1,32 @@
+package it.unibo.authsim.library.dsl
+
+import it.unibo.authsim.library.user.model.UserInformation
+
+/**
+ * Trait for a UserProvider
+ */
+trait UserProvider :
+  def userInformations(): List[UserInformation]
+
+/**
+ * Object for a UserProvider
+ */
+object UserProvider:
+  /**
+   * Apply method for the object
+   *
+   * @return a UserProvider
+   */
+  def apply(): UserProvider= new UserProvider:
+    //NB Non sto a scrivere altro perchè non so quanto durerà sta variabile
+    /**
+     * Variable that represent a representation of a userInformations provided
+     */
+    private var _userInformations: List[UserInformation] = List.empty
+
+    /**
+     * Getter method for the userInformation 
+     * 
+     * @return a List of UserInformation
+     */
+    override def userInformations(): List[UserInformation]= _userInformations
```

Comment:
As UserProvider is a mere "interface", there should be no need to have a companion object, as everybody who wants to use UserProvider should extend it.

If we want to have a companion object, then it can:
* Return an empty List;
* Return a List with 1 element (a "default" element, with trivial values)

I'm for not having a companion object (for the reason above), but we can keep it if you want, as it could provide a simple and fast way to have a working implementation.

Reply:
The companion object, as I had advanced weeks ago when it was implemented, it was needed only for testing reason.
I think that it is better to remove it.